docs/Debugging.md

```
@@ -0,0 +1,65 @@
+### Mocha Tests (Unit, Integration, E2E)
+
+*   **Verbose Output:** Run Mocha with increased verbosity if needed (though the default `spec` reporter is usually detailed). Check `scripts/run-tests.sh` for current settings.
+*   **Debugger:** Use the Node.js inspector:
+    ```bash
+    # Add --inspect-brk flag
+    node --inspect-brk --import 'data:text/javascript,...' ./node_modules/mocha/bin/mocha ... [your test path]
+    ```
+    Then attach your debugger (e.g., Chrome DevTools, VS Code debugger).
+*   **`console.log`:** Add temporary `console.log` statements in the test or the code being tested.
+*   **Mocking Issues (Unit/Integration):**
+    *   Verify mocks (`sinon`, `nock`) are correctly set up and restored (`beforeEach`/`afterEach`).
```

markdownlint-cli2 (0.17.2)

Unordered list indentation
Expected: 2; Actual: 4

(MD007, ul-indent)